src/effects/presets/effectchainpresetmanager.cpp

```diff
@@ -668,10 +695,27 @@ EffectsXmlData EffectChainPresetManager::readEffectsXml(
 
     prependRemainingPresetsToLists();
 
+    // Create the empty '---' chain preset on each start.
+    // Its sole purpose is to eject the current effect chain presets via GUI.
+    // It will not be saved to effects/chains nor written to effects.xml
+    // except as identifier for QuickEffects.chains.
+    // It will not be visible in the effects preferences.
+    EffectManifestPointer pEmptyChainManifest(new EffectManifest());
+    pEmptyChainManifest->setName(kNoEffectString);
+    pEmptyChainManifest->setShortName(kNoEffectString);
+
+    auto pEmptyChainPreset =
+            EffectChainPresetPointer(new EffectChainPreset(pEmptyChainManifest));
+    pEmptyChainPreset->setName(pEmptyChainManifest->name());
```

Comment:
I can also imagine to create a new `class EmptyStaticEffectChainPreset : public EffectChainPreset` to outsource this setup.

If all those `preset->name() == kNoEffectString` tests are not considered safe or "nice" that might also allow to use `static EmptyStaticEffectChainPreset::name` instead, or try to cast to `EmptyStaticEffectChainPreset` where applicable.

Reply:
`EmptyStaticEffectChainPreset` is overkill, I'll try to add `setReadOnly`/`isReadOnly` to `EffectChainPreset`.